### PR TITLE
maintenance/strict-typescript

### PR DESCRIPTION
**Pull request by daviduzumeri**

### CODE REVIEW

File: src/components/Form/index.spec.tsx [hunk 1]
```diff
@@ -72,7 +78,7 @@ describe('Form', () => {
       </ThemeProvider>
     )
 
-    const myForm = container.querySelector('[name="my-form"]')
+    const myForm = container.querySelector('[name="my-form"]')!
```
Comment (clayton-duarte) [inline src/components/Form/index.spec.tsx:81]: I assume that I had to search for the `!` at the end of the expression

Comment (daviduzumeri) [reply src/components/Form/index.spec.tsx:81]: Yeah the `!` just sets it to not-null

File: src/components/InputAutocomplete/index.tsx [hunk 2]
```diff
@@ -55,9 +55,9 @@ export enum Orientation {
 type ParentRect = DOMRectList[0]
 
 interface StyledPopoverProps {
-  orientation: Orientation
-  parent: ParentRect
-  open: boolean
+  orientation?: Orientation
+  parent?: ParentRect
+  open?: boolean
```
Comment (clayton-duarte) [inline src/components/InputAutocomplete/index.tsx:60]: I think only `parent` doesnt have a default value defined in there. Should we default it to 0?

Comment (daviduzumeri) [reply src/components/InputAutocomplete/index.tsx:60]: I mean we're returning undefined if it's not there etc, I think things are fine as is?

File: src/components/InputAutocomplete/index.tsx [hunk 3]
```diff
@@ -200,7 +200,7 @@ const InputAutoComplete: FunctionComponent<InputAutoCompleteProps> = ({
         <StyledPlaceholder value={value}>{placeholder}</StyledPlaceholder>
       )}
       <StyledPopover
-        open={suggestions.length && value && open}
+        open={!!(suggestions.length && value && open)}
```
Comment (clayton-duarte) [inline src/components/InputAutocomplete/index.tsx:203]: question: I'm a personal fan of using `!!` but some of our friends think that `Boolean()` is clearer... What's your opinion?

File: src/components/Palette/index.doc.tsx [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
 
 interface PaletteProps {
   palette?: keyof Theme['palette']
   color?: keyof Theme['colors']
-  theme?: Theme
 }
 
-const returnColor = ({ theme, palette, color }: PaletteProps): string => {
+const returnColor = ({ palette, color }: PaletteProps): string => {
+  const theme = useTheme()
```
Comment (clayton-duarte) [inline src/components/Palette/index.doc.tsx:12]: nice catch

File: src/components/InputAutocomplete/index.tsx (hunk 3 above)
Comment (daviduzumeri) [reply src/components/InputAutocomplete/index.tsx:203]: I'm cool with either? I can switch it over.

File: src/components/SvgIcon/index.spec.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ import iconList from './icons'
 import ThemeProvider from '../..'
 import SvgIcon from '.'
 
-type IconNames = keyof typeof iconList
+const iconNames = Object.keys(iconList) as (keyof typeof iconList)[]
```
Comment (clayton-duarte) [inline src/components/SvgIcon/index.spec.tsx:8]: much more elegant than my previous code that way!

File: tsconfig.json [hunk 6]
```diff
@@ -10,7 +10,7 @@
     "module": "commonjs",
     "jsx": "preserve",
     "allowJs": false,
-    "strict": false,
+    "strict": true,
```
Comment (clayton-duarte) [inline tsconfig.json:13]: I think that worth to write a ticket to do the same on OOF

**Review by clayton-duarte — APPROVED** [review]

File: tsconfig.json (hunk 6 above)
Comment (daviduzumeri) [reply tsconfig.json:13]: Agreed, but I want to continue with the overhaul here first :)

File: src/components/Form/index.spec.tsx (hunk 1 above)
Comment (clayton-duarte) [reply src/components/Form/index.spec.tsx:81]: hehe, looks like a "believe me when I say that's not null!"

File: src/components/InputAutocomplete/index.tsx (hunk 2 above)
Comment (clayton-duarte) [reply src/components/InputAutocomplete/index.tsx:60]: well, `undefinedpx` won't cause any problem actually, CSS will just read it as invalid and ignore it (we don't need to worry about any inheritance issue to the CSS-in-JS usage)

File: src/components/InputAutocomplete/index.tsx (hunk 3 above)
Comment (clayton-duarte) [reply src/components/InputAutocomplete/index.tsx:203]: no, no, `!!` is cool

**Pull request merged**